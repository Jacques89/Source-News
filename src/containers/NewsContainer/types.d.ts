type NewsContainerProps = {
  news: Object<string>
  category: string
  userInput: string
}