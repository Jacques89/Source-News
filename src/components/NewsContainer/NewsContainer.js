import React, { useState, useEffect } from 'react'

import { useSpring, animated } from 'react-spring'

import NewsBox from '../NewsBox/NewsBox'
import "../../styles/NewsContainer.scss"

const NewsContainer = ({ news, category, userInput }) => {

  const [articles, setArticles] = useState('')
  const articleAnimation = useSpring({ opacity: 1, from: { opacity: 0, transitionDelay: '.5s', transitionDuration: '.5s' } })

  useEffect(() => {
    setArticles(
      news
      .filter(article => article.category === category)
      .map((article, i) => 
        <animated.div style={articleAnimation} className="box" key={i}><NewsBox key={i} article={article} /></animated.div>
      )
    )
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [userInput, category, news])
 

  return (
    <div className="container">
      {
        articles &&
        !userInput && articles.length === 0 ?
        `Sorry, there is no news in ${category} currently available`
        : userInput && articles.length === 0 ?
          `Sorry, there is no news currently available to ${userInput} in the ${category} category`
          :
          articles
      }
    </div>
  )
}

export default NewsContainer