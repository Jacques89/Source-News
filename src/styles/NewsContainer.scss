.container {
  width: 100%;
  min-height: 64vh;
  display: grid;
  grid-template-columns: repeat( auto-fit, minmax(300px, 1fr) );
  grid-gap: 1rem;
  padding: 1.5rem;

  @media screen and (max-width: 320px) {
    padding: 0.6rem;
  }
}

.message {
  color: #fff;
}

.no-news {
  position: absolute;
  z-index: -1;
  top: 15.625rem;
  left: 3.125rem;
  font-size: 1rem;
  color: #fff;

  @media screen and (min-width: 320px) {
    margin-top: 40px;
  }


  @media screen and (max-width: 670px) {
    top: 11.25rem;
    left: 1.875rem;
  }

  @media screen and (min-width: 1200px) {
    min-height: 50vh;
  }

}

.box {
  border: 1px solid rgb(150, 150, 150);
  background-color: #fff;
  position: relative;
  padding-bottom: 2rem;
  box-shadow: 0 0 0.313rem grey;

  .link:hover {
    transform: scale(1.1);
  }
}

@media screen and (min-width: 1000px) {
  .box:first-child {
    grid-row: 1/2;
    grid-column: 1/4;
    
    h3 {
      font-size: 2.5rem;
    }

    .image {
      height: 300px;
    }

    p {
      font-size: 1.5rem;
    }

    .link {
      font-size: 1rem;
    }
  }
}