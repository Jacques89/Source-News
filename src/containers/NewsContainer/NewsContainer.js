import React, { useState, useEffect } from 'react'

import { useSpring, animated } from 'react-spring'

import NewsBox from '../../components/NewsBox/NewsBox'
import '../../styles/NewsContainer.scss'

const NewsContainer = ({ news, category, userInput }) => {
  const [articles, setArticles] = useState('')
  const articleAnimation = useSpring({
    opacity: 1,
    from: { opacity: 0, transitionDelay: '.5s', transitionDuration: '.5s' }
  })

  useEffect(() => {
    setArticles(
      news
        .filter(
          (article) =>
            article.title
              .toLocaleLowerCase()
              .indexOf(userInput.toLocaleLowerCase()) !== -1
        )
        .filter((article) => article.category === category)
        .map((article, i) => (
          <animated.div style={articleAnimation} className='box' key={i}>
            <NewsBox key={i} article={article} />
          </animated.div>
        ))
    )
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [userInput, category])

  const noNewsCategory = `Sorry, there is no news in ${category} currently available`
  const noNewsUserInput = `Sorry, there is no news currently available for ${userInput} in ${category}`

  return (
    <div className='container'>
      {articles && !userInput && articles.length === 0 ? (
        <div className='no-news'>{noNewsCategory}</div>
      ) : userInput && articles.length === 0 ? (
        <div className='no-news'>{noNewsUserInput}</div>
      ) : (
        articles
      )}
    </div>
  )
}

export default NewsContainer
