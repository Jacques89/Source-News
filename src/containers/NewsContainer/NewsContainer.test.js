import React from 'react'
import NewsContainer from './NewsContainer'

import { shallow } from 'enzyme'

describe('NewsContainer', () => {
  let wrapper
  beforeEach(() => (
  wrapper = shallow(<NewsContainer />)
))

  it('renders correctly', () => {
    const wrapper = shallow(<NewsContainer />)
  
    expect(wrapper).toMatchSnapshot()
  })

  it('should render a <div>', () => {
    expect(wrapper.find('div').length).toEqual(1)
  })
})
