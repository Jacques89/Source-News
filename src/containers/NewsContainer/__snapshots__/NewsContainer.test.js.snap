// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer renders correctly 1`] = `ShallowWrapper {}`;
